src/CompilerOptions.ts

```diff
@@ -41,3 +43,42 @@ export enum LuaTarget {
     Lua53 = "5.3",
     LuaJIT = "JIT",
 }
+
+export function validateOptions(options: CompilerOptions): ts.Diagnostic[] {
+    const diagnostics: ts.Diagnostic[] = [];
+
+    if (options.luaBundle && (options.luaBundleEntry === "" || !options.luaBundleEntry)) {
+        diagnostics.push(configErrorDiagnostic(`'luaBundleEntry' is required when 'luaBundle' is enabled.`));
+    }
+
+    if (options.luaBundle && options.luaLibImport === LuaLibImportKind.Inline) {
+        diagnostics.push(
+            configWarningDiagnostic(
+                `Using 'luaBundle' with 'luaLibImport: "inline"' might generate duplicate code. ` +
+                    `It is recommended to use 'luaLibImport: "require"'`
+            )
+        );
+    }
+
+    return diagnostics;
+}
+
+const configErrorDiagnostic = (message: string): ts.Diagnostic => ({
```

Comment:
IMO diagnostic factories should be specified, that would look better once we'll have diagnostic codes